SCClassLibrary/Common/Control/Server.sc

```diff
@@ -669,77 +577,91 @@ Server {
 		^Bus(\audio, 0, this.options.numOutputBusChannels, this);
 	}
 
-	startAliveThread { arg delay=0.0;
-		this.addStatusWatcher;
-		^aliveThread ?? {
-			aliveThread = Routine({
-				// this thread polls the server to see if it is alive
-				delay.wait;
-				loop({
-					this.status;
-					aliveThreadPeriod.wait;
-					this.serverRunning = alive;
-					alive = false;
-				});
-			});
-			AppClock.play(aliveThread);
-			aliveThread
-		}
+
+	/* server status */
+
+	numUGens { ^statusWatcher.numUGens }
+	numSynths { ^statusWatcher.numSynths }
+	numGroups { ^statusWatcher.numGroups }
+	numSynthDefs { ^statusWatcher.numSynthDefs }
+	avgCPU { ^statusWatcher.avgCPU }
+	peakCPU { ^statusWatcher.peakCPU }
+	sampleRate { ^statusWatcher.sampleRate }
+	actualSampleRate { ^statusWatcher.actualSampleRate }
+	serverRunning { ^statusWatcher.serverRunning }
+	serverBooting { ^statusWatcher.serverBooting }
+	unresponsive { ^statusWatcher.unresponsive }
+
+	startAliveThread { | delay=0.0 |
+		statusWatcher.startAliveThread(delay)
 	}
 	stopAliveThread {
-		if( aliveThread.notNil, {
-			aliveThread.stop;
-			aliveThread = nil;
-		});
-		if( statusWatcher.notNil, {
-			statusWatcher.free;
-			statusWatcher = nil;
-		});
+		statusWatcher.stopAliveThread
 	}
 	aliveThreadIsRunning {
-		^aliveThread.notNil and: {aliveThread.isPlaying}
+		^statusWatcher.aliveThread.isPlaying
+	}
+
+	aliveThreadPeriod_ { |val|
+		statusWatcher.aliveThreadPeriod_(val)
+	}
+	aliveThreadPeriod { |val|
+		^statusWatcher.aliveThreadPeriod
 	}
+
+
+
+	disconnectSharedMemory {
```

Comment:
this wasn't supposed to be here. I will add a commit to remove it. its part of some other feature